### PR TITLE
add username as a parameter for redis components on 3.4.x

### PR DESCRIPTION
**Pull request by vcidst**
**Proposed changes**:
- Add `username` to the connection parameters for RedisLockStore

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/lock-stores.mdx [hunk 1]
```diff
@@ -156,6 +156,8 @@ The `ConcurrentRedisLockStore` recreates the `TicketLock` from the persisted `Ti
   - `key_prefix` (default: `None`): The prefix to prepend to lock store keys. Must
     be alphanumeric
 
+  -  `username` (default: `None`): Username used for authentication
```
Comment (radovanZRasa) [inline docs/docs/lock-stores.mdx:159]: Should be aligned.

File: tests/core/test_lock_store.py [hunk 2]
```diff
@@ -384,3 +387,42 @@ async def test_redis_lock_store_with_valid_prefix(monkeypatch: MonkeyPatch):
     with pytest.raises(LockError):
         async with lock_store.lock("some sender"):
             pass
+
+
+def test_tracker_store_endpoint_config_loading(endpoints_path: Text):
```
Comment (radovanZRasa) [inline tests/core/test_lock_store.py:392]: Default fixture file does not have field username.
Create a new fixture file and use that.

File: tests/core/test_lock_store.py [hunk 3]
```diff
@@ -384,3 +387,42 @@ async def test_redis_lock_store_with_valid_prefix(monkeypatch: MonkeyPatch):
...
+            "timeout": 30000,
+            "use_ssl": True,
+            "ssl_keyfile": "keyfile.key",
+            "ssl_certfile": "certfile.crt",
+            "ssl_ca_certs": "my-bundle.ca-bundle",
+        }
+    )
+
+
+def test_create_lock_store_from_endpoint_config(domain: Domain, endpoints_path: Text):
```
Comment (radovanZRasa) [inline tests/core/test_lock_store.py:412]: Same here.

File: docs/docs/lock-stores.mdx (hunk 1 above)
Comment (vcidst) [reply docs/docs/lock-stores.mdx]: Fixed it

File: tests/core/test_lock_store.py (hunk 2 above)
Comment (vcidst) [reply tests/core/test_lock_store.py:392]: Ah, I've added the username field in the existing yaml file. I hope that also works?

**Review by radovanZRasa — APPROVED** [review]

**Pull request merged**